thoth/storages/graph/postgres.py

```diff
@@ -2970,6 +2970,66 @@ def python_package_index_listing(self, enabled: bool = None) -> list:
 
         return [{"url": item[0], "warehouse_api_url": item[1], "verify_ssl": item[2]} for item in query.all()]
 
+    def internal_hardware_environments_listing(self) -> list:
+        """Get a list of hardware environments registered in the graph database."""
+        query = self._session.query(
+            HardwareInformation.cpu_cores,
```

Comment:
You can directly issue:

```python
result = self._session.query(HardwareInformation).all()
```

and subsequently transform models returned using:

```python
return [model.to_dict() for model in result]
```

Reply:
Thanks!